deploy-service/common/src/main/java/com/pinterest/deployservice/scm/PhabricatorManager.java

```diff
@@ -41,7 +41,7 @@
 public class PhabricatorManager extends BaseManager {
 
     private static final Logger LOG = LoggerFactory.getLogger(PhabricatorManager.class);
-    public static final String TYPE = "Phabricator";
+    public static String TYPE;
```

Comment:
is TYPE still needed in this class?

Reply:
I kept it to keep the static reference to it in SourceControlManagerProxy. Since Phabricator is the default SCM and I did not want to use a hardcoded string, I assigned TYPE to the value of typeName (in this case "Phabricator" from the config file) and created a static method that returns TYPE since getTypeName() returns a non static variable (and shouldn't because the value of typeName comes from the config file).

Reply:
Thanks, is it only consumed in SourceControlMangerProxy? If so we can change that to support the new non-static usage and reduce the total lines of code

Reply:
Yes and I'm trying to find a way to do that, but it looks like I have to return a static variable for that static method referenced in SourceControlManagerProxy. The getTypeName() method wouldn't work because that's a method that can only be called by instances of the PhabricatorManager class. My understanding of a static method is that it acts as a class method, so if we want to default type to be "Phabricator" we need a way of passing the value of typeName (non-static) to TYPE (static) which I did by initializing TYPE and assigning it to typeName and returning it in the getStaticTypeName() method which is used in SourceControlManagerProxy